cicecore/cicedynB/infrastructure/comm/mpi/ice_communicate.F90

```diff
@@ -13,18 +13,6 @@ module ice_communicate
    use ice_exit, only: abort_ice
    use icepack_intfc, only: icepack_warnings_flush, icepack_warnings_aborted
 
-#if defined key_oasis3 || key_oasis3mct
-   use cpl_oasis3
-#endif
-
-#if defined key_oasis4
-   use cpl_oasis4
-#endif
-
-#if defined key_iomput
-   use lib_mpp, only:   mpi_comm_opa      ! MPP library
-#endif
-
```

Comment:
OK, so basically these `use` statements would have to be in the drivers instead, and the communicator passed to `init_communicate` by argument. I don't know why it was not made that way in the first place, but I think this is a good change that cleans up the code.

Thanks. I'll update my code next time I rebase the branch where I'm working on  coupling with NEMO.

Reply:
That's correct.  I'm actually surprised anyone is using the key_ cpps.  I thought that was obsolete by now.  If not, we probably need to make sure we call this out in our release notes for Oasis users.

Reply:
Yes, it definitely should be in the release notes. Also, `key_iomput` is not OASIS-related, it's plain NEMO. We use it at CMC although we do not use OASIS.

Reply:
Thanks @phil-blain, very interesting.  Do you agree it's reasonable and appropriate to remove these cpps and force users to pass the comm thru the interface?

Reply:
Yes I think it is.